nexus/db-model/src/instance.rs

```diff
@@ -63,6 +63,10 @@ pub struct Instance {
     #[diesel(column_name = auto_restart_policy)]
     pub auto_restart_policy: Option<InstanceAutoRestart>,
 
+    /// The primary boot device for this instance.
+    #[diesel(column_name = boot_device)]
+    pub boot_device: Option<String>,
```

Comment:
if this were just the field it is, it should be `Name`, since it is .. the name of some other boot device. but on the Propolis side we'll actually accept something [more flexible](https://github.com/oxidecomputer/propolis/pull/756#discussion_r1761872738).

@ahl @askfongjojo @karencfv and @david-crespo i'd like your thoughts on how much we should carry that through: recording a `BootOrder` struct which we can then extend in the future is certainly nice for internal purposes, but i'm less sure how to judge the complexity/extensibility tradeoff for the more user-facing API here.

if we have just "boot device" here it seems straightforward in all uses today, but allowing a future boot order list, or "first boot only"-style settings on boot entries, means we'd need a new field and presumably to allow only one of `{boot_device, future_more_complex_list}`.

i'm inclined towards keeping just the single optional `boot_device` string to avoid the extra layers in user-facing APIs, but i don't know if there's other context i be aware of.

Reply:
Due to names being mutable, I'd prefer the ID of the boot device. For users who rely on automation to manage their infrastructure, and immutable identifier is always preferable. Like how we have `project_id` as part of this struct :)

Reply:
Yeah, ID would be much more typical. Often we allow something to be set by NameOrID, but we resolve it to an ID immediately and store it as an ID.

Reply:
oh! i wasn't sure that they were mutable. in that case, yes, the ID would be much better. it turns out the name is what gets sent out to Propolis, so i'd just stuck with that here, but perhaps Propolis should be getting IDs as well.

Reply:
I'd definitely advocate for changing the Propolis API to take a UUID, if at all possible.

Reply:
when we're creating the instance, we require that [`InstanceDiskAttach` specify a name](https://github.com/oxidecomputer/omicron/blob/main/nexus/types/src/external_api/params.rs#L948-L952), and not an ID. we probably ought to take `NameOrId` there. otherwise allowing both here seems to be of limited utility.. the use cases afaict look like:

* if you're creating new disks along with the instance creation, you don't have IDs yet, so you couldn't provide an ID to `boot_disk`. simple enough, strong indicator we should at least take Name here.
* if you're attaching existing disks, you _must_ use their Name, so passing a disk ID here seems like net-more work than just reusing the name you'd have put in `disks[]` already.
* if you're updating the `boot_disk` of a stopped instance, that seems like the most likely case for an ID on its own to be especially helpful.

i don't want to go change `InstanceDiskAttach` here too (basically just because i don't know if that is laden with secret cans of worms), but if folks agree it seems like it would be nice to allow NameOrId sooner than later for disk attachment.

Reply:
Yeah, we did intend to change that, but I never got around to wrapping it up after Justin left:

https://github.com/oxidecomputer/omicron/pull/3671

Reply:
> but if folks agree it seems like it would be nice to allow NameOrId sooner than later for disk attachment.

Definitely on board with this!